syntax = "proto3";

import "transaction.proto";
import "ca_protomsg.proto";
import "sign.proto";


message CBlock {
  uint32 version = 1;
  uint64 time = 2;
  string hash = 3;
  string prevHash = 4;
  int64 height = 5;
  string merkleRoot = 6;
  repeated CTransaction txs = 7;
  string data = 8;
  string info = 9;

  repeated CSign sign = 10;

  string reserve0 = 11;
  string reserve1 = 12;
}

message BlockMsg
{
  string    version             = 1;
  sint32    code                = 2;
  string    message             = 3;
  uint64    time                = 4;
  bytes     block               = 5;
  repeated Vrf  vrfInfo      = 6;
  repeated Vrf  txvrfInfo    = 7;
  VrfDataSource vrfDataSource   = 8;
  CSign         sign            = 9;
}

message TxStatus
{
  string  txHash      = 1;
  int32   status      = 2;
}

message BlockStatus
{
  string              blockHash   = 1;
  int32               status      = 2;
  repeated TxStatus   txStatus    = 3;
  string              id          = 4;
}

message SeekPreHashByHightReq
{
    string                  self_node_id         = 1;
    string                  msg_id               = 2;
    uint64 		              seek_height          = 3;
}

message SeekPreHashByHightAck
{
    string                  self_node_id       = 1;
    string                  msg_id             = 2;
    uint64 		              seek_height        = 3;
    repeated string         prehashes          = 4;
}

message SeekContractPreHashReq
{
    string                  self_node_id         = 1;
    string                  msg_id               = 2;
}

message SeekContractPreHashAck
{
    string                  self_node_id       = 1;
    string                  msg_id             = 2;
    string                  ContractPreHash    = 3;
}


//--
message preHashPair
{
  string ContractAddr = 1;
  string RootHash = 2;
  bytes blockRaw = 3;
}


message newSeekContractPreHashReq
{
  string version = 1;
  string msg_id  = 2;
  string self_node_id = 3;
  repeated preHashPair seekRootHash = 4;
}

message newSeekContractPreHashAck
{
  string version = 1;
  string msg_id  = 2;
  string self_node_id = 3;
  repeated preHashPair seekContractBlock = 4;
}
